rewrite-java/src/main/java/org/openrewrite/java/cleanup/UseJavaInterface.java

```diff
@@ -0,0 +1,185 @@
+        rspecRulesReplaceTypeMap.put("java.util.HashMap", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.Hashtable", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.IdentityHashMap", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.LinkedHashMap", "java.util.Map");
+        rspecRulesReplaceTypeMap.put("java.util.WeakHashMap", "java.util.Map");
+        // ConcurrentMap
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.ConcurrentHashMap", "java.util.concurrent.ConcurrentMap");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.ConcurrentSkipListMap", "java.util.concurrent.ConcurrentMap");
+        // Queue
+        rspecRulesReplaceTypeMap.put("java.util.AbstractQueue", "java.util.Queue");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.ConcurrentLinkedQueue", "java.util.Queue");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.SynchronousQueue", "java.util.Queue");
+        // Set
+        rspecRulesReplaceTypeMap.put("java.util.AbstractSet", "java.util.Set");
+        rspecRulesReplaceTypeMap.put("java.util.EnumSet", "java.util.Set");
+        rspecRulesReplaceTypeMap.put("java.util.HashSet", "java.util.Set");
+        rspecRulesReplaceTypeMap.put("java.util.LinkedHashSet", "java.util.Set");
+        rspecRulesReplaceTypeMap.put("java.util.concurrent.CopyOnWriteArraySet", "java.util.Set");
+        // Sorted
+        rspecRulesReplaceTypeMap.put("java.util.TreeMap", "java.util.SortedMap");
+        rspecRulesReplaceTypeMap.put("java.util.TreeSet", "java.util.SortedSet");
+
+        return new JavaVisitor<ExecutionContext>() {
+
+            @Override
+            public J visitCompilationUnit(J.CompilationUnit cu, ExecutionContext executionContext) {
+                for (JavaType type : cu.getTypesInUse()) {
+                    if (type instanceof JavaType.Class && rspecRulesReplaceTypeMap.containsKey(((JavaType.Class) type).getFullyQualifiedName())) {
+                        return super.visitCompilationUnit(cu, executionContext);
+                    }
+                }
+
+                return cu;
+            }
+
+            @Override
+            public J visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext executionContext) {
+                J.MethodDeclaration m = (J.MethodDeclaration) super.visitMethodDeclaration(method, executionContext);
+                if (m.hasModifier(J.Modifier.Type.Public) && m.getReturnTypeExpression() != null) {
+                    JavaType.FullyQualified fq = TypeUtils.asFullyQualified(m.getReturnTypeExpression().getType());
+                    if (fq != null && rspecRulesReplaceTypeMap.containsKey(fq.getFullyQualifiedName())) {
+                        JavaType.FullyQualified newType = TypeUtils.asFullyQualified(
+                                JavaType.buildType(rspecRulesReplaceTypeMap.get(fq.getFullyQualifiedName())));
```

Comment:
`JavaType.Class.find()` should always be able to successfully locate the existing types for java.lang built-ins like these. Might as well use that.